ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -128,39 +128,49 @@ public synchronized void add(final ValidatableAttestation attestation) {
     }
   }
 
+  private Optional<Int2IntMap> retrieveCommitteesSize(final Attestation attestation) {
+    if (attestation.requiresCommitteeBits()) {
+      return getCommitteesSizeSupplierUsingTheState(attestation.getData());
+    }
+    return Optional.empty();
+  }
+
+  private boolean attestationIsApplicable(
+      final Attestation attestation, final Optional<Int2IntMap> committeesSize) {
+    if (attestation.requiresCommitteeBits()) {
+      // if an attestation has committee bits, committees size should have been computed. If this is
+      // not the case, we can ignore this attestation and not add it to the pool
+      return committeesSize.isPresent();
+    }
+    return true;
+  }
+
   /**
-   * @param committeesSizeSupplier Required for aggregating attestations as per <a
+   * @param committeesSize Required for aggregating attestations as per <a
    *     href="https://eips.ethereum.org/EIPS/eip-7549">EIP-7549</a>
    */
   private MatchingDataAttestationGroup getOrCreateAttestationGroup(
-      final AttestationData attestationData, final Supplier<Int2IntMap> committeesSizeSupplier) {
+      final AttestationData attestationData, final Optional<Int2IntMap> committeesSize) {
     dataHashBySlot
         .computeIfAbsent(attestationData.getSlot(), slot -> new HashSet<>())
         .add(attestationData.hashTreeRoot());
     return attestationGroupByDataHash.computeIfAbsent(
         attestationData.hashTreeRoot(),
-        key -> new MatchingDataAttestationGroup(spec, attestationData, committeesSizeSupplier));
+        key -> new MatchingDataAttestationGroup(spec, attestationData, committeesSize));
   }
 
   // We only have the committees size already available via attestations received in the gossip
   // flow and have been successfully validated, so querying the state is required for other cases
-  private Supplier<Int2IntMap> getCommitteesSizeSupplierUsingTheState(
+  private Optional<Int2IntMap> getCommitteesSizeSupplierUsingTheState(
       final AttestationData attestationData) {
-    return () -> {
-      final Bytes32 targetRoot = attestationData.getTarget().getRoot();
-      LOG.debug(
-          "Committees size was not readily available for attestation with target root {}. Will attempt to retrieve it using the relevant state.",
-          targetRoot);
-      final BeaconState state =
-          recentChainData
-              .getStore()
-              .getBlockStateIfAvailable(targetRoot)
-              .orElseThrow(
-                  () ->
-                      new IllegalStateException(
-                          "No state available for attestation with target root " + targetRoot));
-      return spec.getBeaconCommitteesSize(state, attestationData.getSlot());
-    };
+    final Bytes32 targetRoot = attestationData.getTarget().getRoot();
+    LOG.debug(
+        "Committees size was not readily available for attestation with target root {}. Will attempt to retrieve it using the relevant state.",
```

Comment:
i'd probably change this to just be something short like
`Committee size was not found via target root {} for attestation at slot {}`